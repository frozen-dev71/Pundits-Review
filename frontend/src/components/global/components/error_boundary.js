// ======================================
// Error Boundary Component - Redirects Users to Error Page if error is caught
// Wraps components making API calls 
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component, Fragment } from "react";
import { render } from "react-dom";
import { Link, withRouter } from 'react-router-dom';
import { connect } from "react-redux";
import { resetErrorBoundary } from '../../actions/actions';


// Local JS Imports 



// ===========
// Error Boundary Component 
// ===========


export class ErrorBoundaryWrapper extends Component {

	componentDidUpdate() {

		if (this.props.errorExists && this.props.history.location.pathname !== "/error") {

			this.props.history.push("/error");

			// Reset Error Boundary
			this.props.resetErrorBoundary();
		}

	}

	render() {

		return <span>{this.props.children}</span>;
	}
}



// ===========
// Connect component to redux store
// ===========

const mapStateToProps = state => {
  return { 
  		errorExists: state.errorExists
  	};
};


export const ErrorBoundary = withRouter(connect(mapStateToProps, { resetErrorBoundary } )(ErrorBoundaryWrapper));

