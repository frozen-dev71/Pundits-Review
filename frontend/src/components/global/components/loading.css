/*// ======================================
// LOADING SCREEN CSS - contains CSS for the icon and text to show user that page is loading
// ======================================*/


.loading-screen-wrapper {
	display: flex;
	flex-direction: column;
	justify-content: space-around;
	text-align: center;
	height: 50vh;
	width: 100vw;
}

.loading-screen-text {
	font-size: 2em;
	letter-spacing: .25em;
}