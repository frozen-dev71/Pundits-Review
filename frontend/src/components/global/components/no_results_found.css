/*
// ======================================
// CSS for no results component 
// ======================================
*/

.no-results-wrapper {
	display: flex;
	flex-direction: column;
    text-align: center;
    padding: 2em;
    align-items: center;
}

.empty-page-wrapper {
	justify-content: center;
    height: 60vh;
    display: flex;
	flex-direction: column;
    text-align: center;
    padding: 2em;
    align-items: center;
}

.no-results-text {
	font-size: 1.25em;
	letter-spacing: .1em;
	padding: 1em 0;
}

.no-results-sub-text {
	font-size: .85em;
	letter-spacing: .05em;
	padding: 1em 0; 
}


.empty-page-home-link {
	color: black;
	font-size: 1.25em;
	padding: 1em;
	cursor: pointer;
	letter-spacing: .2em;
}

/*
// ======================================
// MEDIA QUERIES
// ======================================
*/


@media (min-width: 1000px) {

	.no-results-text, .empty-page-home-link  {
		font-size: 1.5em;
		padding: 2em 0;
	}

	.no-results-sub-text {
		font-size: 1.15em;
		width: 50vw;
		padding: 2em 0;
	}

}