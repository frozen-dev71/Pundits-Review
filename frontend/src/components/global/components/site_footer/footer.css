/*
// ======================================
// CSS for Site Footer 
// ======================================
*/


.footer-wrapper {
    display: flex;
    flex-direction: column;
    height: 20vh;
    width: 60vw;
    justify-content: center;
    align-items: center;
    bottom: 0;
    margin: 2em 0;
}

.footer-logo-wrapper {
	display: flex;
	justify-content: center;
    align-items: center;
}

.media-icon {
    font-size: 2em;
    padding: .5em;
    color: black;
}

.footer-logo {
	font-size: 3em;
	padding: .5em;
    color: black;
}

.footer-copyright-notice {
	opacity: .5em;
	font-size: .5em;
	letter-spacing: .2em;
	margin: .5em 0;
}


.footer-title {
    font-size: .75em;
    letter-spacing: .2em;
}