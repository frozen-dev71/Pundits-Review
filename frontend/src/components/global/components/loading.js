// ======================================
// LOADING SCREEN component - contains icon and text to show user that page is loading
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// CSS Imports
import './loading.css';

// ===========
// Loading Component 
// ===========

export class Loading extends Component {


	render() {

		let result = (

			 <div className="loading-screen-wrapper" >

			 	<p className="loading-screen-text body-text-light">LOADING</p>

			 </div>

			);


		return result;
	}
}