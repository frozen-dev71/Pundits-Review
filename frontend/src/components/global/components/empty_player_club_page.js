// ======================================
// Empty Player Club Page Component - Shows user that no results were found for their query
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports
import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from 'react-router-dom';

// Local JS Imports 
import { DividingLine } from './dividing_line';

// CSS Imports 
import './no_results_found.css';

// ===========
// No Results Component 
// ===========

export class EmptyPlayerClubPage extends Component {

	render() {

		let result = (

				<div className="empty-page-wrapper">

					<p className="no-results-text body-text-light">It looks like you haven't selected a {this.props.playerOrClub} </p>

					<DividingLine />

					<p className="no-results-sub-text body-text-light">Head back to the homepage to make a new search for a {this.props.playerOrClub}</p>

					<Link to="/" className="react-link empty-page-home-link body-text-light">
						HOME 
					</Link>

				</div>

			);


		return result;
	}
}
