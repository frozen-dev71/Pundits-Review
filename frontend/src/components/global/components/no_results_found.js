// ======================================
// No Results Component - Shows user that no results were found for their query
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports
import React, { Component } from "react";
import { connect } from "react-redux";

// Local JS Imports 
import { DividingLine } from './dividing_line';

// CSS Imports 
import './no_results_found.css';

// ===========
// No Results Component 
// ===========

export const NoResultsWrapper = props => {

	let searchQueryText;

	if (props.searchQuery) {
		searchQueryText = props.searchQuery
	} else {
		searchQueryText = " "
	}

	let subText = "Although we are always trying to collect more reviews, it is worth remembering that we don't always have data on some " + props.playerOrClub + "."

	if (props.thisWeekEveryoneToggle === "THIS WEEK") {

		subText = "It doesn't look like we have data this week for '" + searchQueryText + "'. Try looking under 'ALL WEEKS' to see if we have older results matching your search."
	}

	let result = (

			<div className="no-results-wrapper">

				<p className="no-results-text body-text-light">We couldn't find any {props.playerOrClub} matching "{searchQueryText}"</p>

				<DividingLine />

				<p className="no-results-sub-text body-text-light">{subText}</p>

			</div>

		);


	return result;
}


// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return { 
  		thisWeekEveryoneToggle: state.thisWeekEveryoneToggle,
  		
  	};
};


export const NoResults = connect(mapStateToProps)(NoResultsWrapper);
