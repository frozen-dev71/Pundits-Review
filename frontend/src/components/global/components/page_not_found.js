// ======================================
// Page Not Found Component - Displays text showing the user that a page was not found
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports
import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

// Local JS Imports 
import { DividingLine } from './dividing_line';

// CSS Imports 
import './no_results_found.css';

// ===========
// PageNotFound Component 
// ===========

export class PageNotFound extends Component {

	render() {

		let result = (

				<div className="empty-page-wrapper">

					<Helmet>
				
						<title>Pundits Review - Page Not Found</title>
						<meta property="og:description" content="Nothing to see here ..."/>
						<meta property="og:url" content="https://www.punditsreview.com/"/>
			    		<meta property="og:type" content="football blog data"/>

			    	</Helmet>

					<p className="no-results-text body-text-light">URMM ... NOTHING TO SEE HERE</p>

					<DividingLine />

					<p className="no-results-sub-text body-text-light">Tip: Use the filter on the <Link to="/playerrankings" className="react-link black-text opaque-link">player rankings page</Link> to filter players by club, nationality & position.</p>

					<Link to="/" className="react-link empty-page-home-link body-text-light">
						HOME 
					</Link>

				</div>

			);


		return result;
	}
}
