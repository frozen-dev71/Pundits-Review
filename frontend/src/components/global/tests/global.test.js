// ======================================
// This file tests for the components listed in the GLOBAL sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that EmptyPlayerClubPage renders correctly
// 2. Check that ErrorBoundary renders correctly
// 3. Check that Loading renders correctly
// 4. Check that NoResults renders correctly
// 5. Check that PageNotFound renders correctly
// 6. Check that Hamburger renders correctly
// 7. Check that SideMenu renders correctly
// 8. Check that TopMenu renders correctly
// 9. Check that Footer renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { EmptyPlayerClubPage } from '.././components/empty_player_club_page';
import { ErrorBoundary } from '.././components/error_boundary';
import { Loading } from '.././components/loading';
import { NoResults } from '.././components/no_results_found';
import { PageNotFound } from '.././components/page_not_found';

import { Hamburger } from '.././components/top_menu/hamburger';
import { SideMenu } from '.././components/top_menu/side_menu';
import { TopMenu } from '.././components/top_menu/top_menu_component';

import { Footer } from '.././components/site_footer/footer';

import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Global Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that EmptyPlayerClubPage renders correctly
	// ===========

	describe('EmptyPlayerClubPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><EmptyPlayerClubPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that ErrorBoundary renders correctly
	// ===========

	// describe('ErrorBoundary', () => {

	//   	it('should render correctly', () => {

	// 	  	// Declare component
	// 	  	const component = (<BrowserRouter><Provider store={store}><ErrorBoundary/></Provider></BrowserRouter>);
			
	// 		//
	// 		// Simulate Component
	// 	    const wrapper = shallow(component);

	// 	    // Run Test Case
	// 	    expect(toJson(wrapper)).toMatchSnapshot();

	//   	});

	// })


	// ===========
	// TEST 3. Check that Loading renders correctly
	// ===========

	describe('Loading', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><Loading/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that NoResults renders correctly
	// ===========

	describe('NoResults', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><NoResults/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that PageNotFound renders correctly
	// ===========

	describe('PageNotFound', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PageNotFound/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 6. Check that Hamburger renders correctly
	// ===========

	describe('Hamburger', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><Hamburger/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 7. Check that SideMenu renders correctly
	// ===========

	describe('SideMenu', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SideMenu/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 8. Check that TopMenu renders correctly
	// ===========

	describe('TopMenu', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><TopMenu/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})




	// ===========
	// TEST 9. Check that Footer renders correctly
	// ===========

	describe('Footer', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><Footer/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})





});
