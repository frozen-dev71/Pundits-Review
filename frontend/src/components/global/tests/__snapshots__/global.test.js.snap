// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Component Tests EmptyPlayerClubPage should render correctly 1`] = `
<div
  class="empty-page-wrapper"
>
  <p
    class="no-results-text body-text-light"
  >
    It looks like you haven't selected a  
  </p>
  <div
    class="dividing-line background-dark"
  />
  <p
    class="no-results-sub-text body-text-light"
  >
    Head back to the homepage to make a new search for a 
  </p>
  <a
    class="react-link empty-page-home-link body-text-light"
    href="/"
  >
    HOME
  </a>
</div>
`;

exports[`Global Component Tests Footer should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Footer />
  </Provider>
</Router>
`;

exports[`Global Component Tests Hamburger should render correctly 1`] = `
<div
  class="hamburger-container"
  role="button"
  tabindex="0"
>
  <span
    class="hamburger-bar background-light"
    id="top-hamburger-bar"
  />
  <span
    class="hamburger-bar background-light"
    id="middle-hamburger-bar"
  />
  <span
    class="hamburger-bar background-light"
    id="bottom-hamburger-bar"
  />
</div>
`;

exports[`Global Component Tests Loading should render correctly 1`] = `
<div
  class="loading-screen-wrapper"
>
  <p
    class="loading-screen-text body-text-light"
  >
    LOADING
  </p>
</div>
`;

exports[`Global Component Tests NoResults should render correctly 1`] = `
<div
  class="no-results-wrapper"
>
  <p
    class="no-results-text body-text-light"
  >
    We couldn't find any  matching " "
  </p>
  <div
    class="dividing-line background-dark"
  />
  <p
    class="no-results-sub-text body-text-light"
  >
    It doesn't look like we have data this week for ' '. Try looking under 'ALL WEEKS' to see if we have older results matching your search.
  </p>
</div>
`;

exports[`Global Component Tests PageNotFound should render correctly 1`] = `
<div
  class="empty-page-wrapper"
>
  <p
    class="no-results-text body-text-light"
  >
    URMM ... NOTHING TO SEE HERE
  </p>
  <div
    class="dividing-line background-dark"
  />
  <p
    class="no-results-sub-text body-text-light"
  >
    Tip: Use the filter on the 
    <a
      class="react-link black-text opaque-link"
      href="/playerrankings"
    >
      player rankings page
    </a>
     to filter players by club, nationality & position.
  </p>
  <a
    class="react-link empty-page-home-link body-text-light"
    href="/"
  >
    HOME
  </a>
</div>
`;

exports[`Global Component Tests SideMenu should render correctly 1`] = `
<div
  class="background-dark"
  id="side-menu-container"
>
  <div
    class="side-menu-text-container"
  >
    <a
      class="body-text-light side-menu-text white-text react-link"
      href="/"
    >
      HOME
    </a>
    <a
      class="body-text-light side-menu-text white-text react-link"
      href="/playerrankings"
    >
      PLAYER RANKINGS
    </a>
    <a
      class="body-text-light side-menu-text white-text react-link"
      href="/howitworks"
    >
      HOW IT WORKS
    </a>
    <a
      class="body-text-light side-menu-text white-text react-link"
      href="/blog"
    >
      PUNDITS REVIEW BLOG
    </a>
  </div>
  <div
    class="side-menu-logo-container"
  >
    <a
      class="heading-text side-menu-logo white-text react-link"
      href="/"
    >
      PR
    </a>
  </div>
</div>
`;

exports[`Global Component Tests TopMenu should render correctly 1`] = `
<div
  class="nav-wrapper"
>
  <div
    class="background-dark"
    id="side-menu-container"
  >
    <div
      class="side-menu-text-container"
    >
      <a
        class="body-text-light side-menu-text white-text react-link"
        href="/"
      >
        HOME
      </a>
      <a
        class="body-text-light side-menu-text white-text react-link"
        href="/playerrankings"
      >
        PLAYER RANKINGS
      </a>
      <a
        class="body-text-light side-menu-text white-text react-link"
        href="/howitworks"
      >
        HOW IT WORKS
      </a>
      <a
        class="body-text-light side-menu-text white-text react-link"
        href="/blog"
      >
        PUNDITS REVIEW BLOG
      </a>
    </div>
    <div
      class="side-menu-logo-container"
    >
      <a
        class="heading-text side-menu-logo white-text react-link"
        href="/"
      >
        PR
      </a>
    </div>
  </div>
  <div
    class="top-menu-container background-dark top-menu-height"
  >
    <div
      class="hamburger-container"
      role="button"
      tabindex="0"
    >
      <span
        class="hamburger-bar background-light"
        id="top-hamburger-bar"
      />
      <span
        class="hamburger-bar background-light"
        id="middle-hamburger-bar"
      />
      <span
        class="hamburger-bar background-light"
        id="bottom-hamburger-bar"
      />
    </div>
    <a
      class="react-link"
      href="/"
    >
      <p
        class="white-text heading-text logo"
      >
        Pundits Review
      </p>
    </a>
  </div>
</div>
`;
